components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java

```diff
@@ -80,8 +80,8 @@ public Response resendCodePost(ResendCodeRequestDTO resendCodeRequestDTO) {
         if (notificationResponseBean == null) {
             ErrorDTO errorDTO = new ErrorDTO();
             errorDTO.setRef(Utils.getCorrelation());
-            errorDTO.setMessage("This service is not yet implemented.");
-            return Response.status(Response.Status.NOT_IMPLEMENTED).entity(errorDTO).build();
+            errorDTO.setMessage("User recovery data is not found. Please re-initiate the recovery flow.");
+            return Response.status(Response.Status.BAD_REQUEST).entity(errorDTO).build();
```

Comment:
Will this change only solve the actual issue? If a user wants to get a recovery code but old confirmation code is cleaned up using a clean-up task then will the user get a recovery code now?

Reply:
With the resend-code API, the user will not get a code back again. But the org admin can reinitiate the pwd reset flow again for the user which will eventually initiate another recovery flow.